File: src/domain/summary/minimal.ts
import { Event } from '../events/types';
import { TEXT_CONSTANTS } from '../../shared/constants';

export class MinimalSummaryGenerator {
  static generate(llmEvents: Event[], stagedFiles: string[], diffStats: string): string {
    const lines: string[] = [];
    lines.push('---');
    lines.push('AI-Context (sayu)');
    
    if (stagedFiles.length > 0) {
      const fileList = stagedFiles.slice(0, TEXT_CONSTANTS.MAX_FILE_DISPLAY).join(', ');
      const more = stagedFiles.length > TEXT_CONSTANTS.MAX_FILE_DISPLAY ? ` (+${stagedFiles.length - TEXT_CONSTANTS.MAX_FILE_DISPLAY} more)` : '';
      lines.push(`Files: ${fileList}${more}`);
    }
    
    if (llmEvents.length > 0) {
      const tools = new Set(llmEvents.map(e => e.meta?.tool).filter(Boolean));
      const toolList = Array.from(tools).join(', ') || 'unknown';
      lines.push(`Events: ${llmEvents.length} LLM interactions (${toolList})`);
    } else {
      lines.push(`Events: Code changes only`);
    }
    
    lines.push('---');
    
    return lines.join('\n');
  }
}

File: domain/summary/minimal.py
"""Minimal summary generator for fallback cases"""

from typing import List

from domain.events.types import Event


class MinimalSummaryGenerator:
    """Generate minimal summaries when LLM is unavailable"""
    
    @staticmethod
    def generate(llm_events: List[Event], staged_files: List[str], diff_stats: str) -> str:
        """Generate minimal summary"""
        lines = []
        lines.append('---')
        lines.append('Thought---')
        
        # File information
        if staged_files:
            max_display = 3
            file_list = ', '.join(staged_files[:max_display])
            more = f" (+{len(staged_files) - max_display} more)" if len(staged_files) > max_display else ""
            lines.append(f"Files: {file_list}{more}")
        
        # Event information
        if llm_events:
            # Extract tools from meta
            tools = set()
            for event in llm_events:
                if event.meta and 'tool' in event.meta:
                    tools.add(event.meta['tool'])
            
            tool_list = ', '.join(tools) if tools else 'unknown'
            lines.append(f"Events: {len(llm_events)} LLM interactions ({tool_list})")
        else:
            lines.append("Events: Code changes only")
        
        lines.append('---')
        
        return '\n'.join(lines)
